app/models/concerns/csv_queryable.rb

```diff
@@ -0,0 +1,56 @@
+require 'csv'
+
+module CsvQueryable
+  extend ActiveSupport::Concern
+
+  included do
+    attr_reader :row
+  end
+
+  def initialize(row:)
+    @row = row
+  end
+
+  def ==(other)
+    other.is_a?(self.class) && other.row == row
+  end
+  alias === ==
+  alias eql? ==
+
+  def hash
+    [self.class, row].hash
+  end
+
+  class_methods do
```

Comment:
Much nicer than manually extending the module to add class methods as in the HMCTS' code 👍

Reply:
Yeah, I believe they did it that way as that is the non-rails way of doing it, without using `ActiveSupport::Concern`. Usually is done in gems etc, that are not assuming you will use the gem only with Rails, you might use it with Sinatra too, for example.

But in our case, there is really no point in depriving us from using Rails modules and makes for simpler and sometimes more readable code too.